File: app/src/main/java/com/naosim/rpgsample/ItemListAdapter.java
package com.naosim.rpgsample;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import com.naosim.rpgmodel.lib.value.Item;

import java.util.List;

public class ItemListAdapter<T extends Item> extends ArrayAdapter<T> {

    public ItemListAdapter(Context context, List<T> objects) {
        super(context, 0, objects);
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if (convertView == null) {
            convertView = LayoutInflater.from(getContext()).inflate(R.layout.view_item_row, null);
        }

        T item = getItem(position);
        ((TextView) convertView.findViewById(R.id.itemTextView)).setText(item.getItemName().getValue());

        return convertView;
    }
}

File: app/src/main/kotlin/com/naosim/rpgmodel/lib/android/ItemListAdapter.kt
package com.naosim.rpgmodel.lib.android

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView
import com.naosim.rpgmodel.lib.model.value.Item
import com.naosim.rpgsample.R

class ItemListAdapter<T : Item>(context: Context, objects: List<T>) : ArrayAdapter<T>(context, 0, objects) {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var convertView = convertView
        if (convertView == null) {
            convertView = LayoutInflater.from(context).inflate(R.layout.view_item_row, null)
        }

        val item = getItem(position)
        (convertView!!.findViewById(R.id.itemTextView) as TextView).text = item.itemName.value

        return convertView
    }
}
